modules/social_features/social_group/src/GroupContentVisibilityUpdate.php

```diff
@@ -0,0 +1,205 @@
+    // Load all the GroupContentEntities from Post to Memberships to content.
+    $entities = $group->getContentEntities();
+
+    $posts = self::getPostsFromGroup($group);
+    foreach ($posts as $pid => $post) {
+      $entities[] = $post;
+    }
+    $memberships = $group->getMembers();
+    foreach ($memberships as $member) {
+      $entities[] = $member;
+    }
+
+    $num_operations = count($entities);
+    $operations = [];
+
+    // Update Memberships & ContentVisibility.
+    // As per documentation each entity has it's own operation.
+    for ($i = 0; $i < $num_operations; $i++) {
+      $operations[] = [
+        '\Drupal\social_group\GroupContentVisibilityUpdate::updateVisibility',
+        [
+          $entities[$i],
+          $new_type,
+        ],
+      ];
+    }
+
+    // Provide all the operations and the finish callback to our batch.
+    $batch = [
+      'title' => t('Updating Group Content...'),
+      'operations' => $operations,
+      'finished' => '\Drupal\social_group\GroupContentVisibilityUpdate::updateVisibilityFinishedCallback',
+    ];
+
+    batch_set($batch);
+
+    $group->set('type', $new_type);
+    $group->save();
+  }
+
+  /**
+   * Update visibility for all Group Content based on a new group type.
+   *
+   * @param \Drupal\node\Entity\Node|\Drupal\social_post\Entity\Post|\Drupal\group\GroupMembership|\Drupal\group\Entity\GroupContentInterface $entity
+   *   The content we are updating.
+   * @param string $new_type
+   *   The new Group type.
+   * @param array $context
+   *   Passed on by reference.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function updateVisibility($entity, $new_type, array &$context) {
+    // Find the corresponding visibility for the new group_type.
+    $default_visibility = SocialGroupHelperService::getDefaultGroupVisibility($new_type);
+
+    // Store some results for post-processing in the 'finished' callback.
+    // The contents of 'results' will be available as $results in the
+    // 'finished' function updateVisibilityFinishedCallback().
+    if ($entity instanceof Post) {
+      $entity->setVisibility($default_visibility);
+      $entity->save();
+    }
+    if ($entity instanceof Node) {
+      $entity->set('field_content_visibility', $default_visibility);
+      $entity->save();
+    }
+    // For GroupMembers we have to update the GroupContent.
+    if ($entity instanceof GroupMembership) {
+      $new_group_type = $new_type . '-group_membership';
+      $membershipEntity = $entity->getGroupContent();
+      $membershipEntity->set('type', $new_group_type);
+      $membershipEntity->save();
+      $entity = $membershipEntity;
+    }
+
+    // Add referenced entity to results. Might want to add it to the result.
+    $context['results'][] = $entity;
+
+    // Optional message displayed under the progressbar.
+    $context['message'] = t('Updating Entity ID "@id"', ['@id' => $entity->id()]);
```

Comment:
```suggestion
    $context['message'] = t('Updating group content (@id)', ['@id' => $entity->id()]);
```

I don't think Entity ID means much to users.